src/test/java/uk/gov/companieshouse/accountsDates/accountsDatesTest.java

```diff
@@ -0,0 +1,274 @@
+		assertEquals("2017-03-05", stringDate);
+
+		LocalDate testDate2 = LocalDate.parse("2015-12-17", formatter);
+		String stringDate2 = datesHelper.convertDateToString(testDate2);
+		assertEquals("2015-12-17", stringDate2);
+
+	}
+
+	@Test
+	public void calculatePeriodRangeOver2Years() throws ParseException {
+
+		// Days difference over 2 years
+		LocalDate testDateStart = LocalDate.parse("2013-03-07");
+		LocalDate testDateEnd = LocalDate.parse("2017-12-17");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateStart, testDateEnd, false);
+		assertEquals("57 months", periodMap.get(PERIOD_START));
+		assertEquals("17 December 2017", periodMap.get(PERIOD_END));
+	}
+
+	@Test
+	public void calculatePeriodRangeOver380() throws ParseException {
+
+		// Days difference over 380
+		LocalDate testDateStart = LocalDate.parse("2015-03-07");
+		LocalDate testDateEnd = LocalDate.parse("2017-12-17");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateStart, testDateEnd, false);
+		assertEquals("33 months", periodMap.get(PERIOD_START));
+		assertEquals("17 December 2017", periodMap.get(PERIOD_END));
+	}
+
+	@Test
+	public void calculatePeriodRangeBelow350() throws ParseException {
+
+		// Days difference below 350
+		LocalDate testDateStart = LocalDate.parse("2016-08-03");
+		LocalDate testDateEnd = LocalDate.parse("2017-02-16");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateStart, testDateEnd, false);
+		assertEquals("6 months", periodMap.get(PERIOD_START));
+		assertEquals("16 February 2017", periodMap.get(PERIOD_END));
+	}
+
+	@Test
+	public void calculatePeriodLessThanOneMonth() throws ParseException {
+
+		// Month difference less than one month
+		LocalDate testDateStart = LocalDate.parse("2015-03-07");
+		LocalDate testDateEnd = LocalDate.parse("2015-04-01");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateStart, testDateEnd, false);
+		assertEquals("1 month", periodMap.get(PERIOD_START));
+		assertEquals("1 April 2015", periodMap.get(PERIOD_END));
+	}
+
+	@Test
+	public void calculatePeriodRangeBelow350RoundUp() throws ParseException {
+
+		// Days difference below 350 with month round up ( remaining days > 15 )
+		LocalDate testDateStart = LocalDate.parse("2016-08-01");
+		LocalDate testDateEnd = LocalDate.parse("2017-02-17");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateStart, testDateEnd, false);
+		assertEquals("7 months", periodMap.get(PERIOD_START));
+		assertEquals("17 February 2017", periodMap.get(PERIOD_END));
+	}
+
+	@Test
+	public void calculatePeriodRangeSameYear() throws ParseException {
+		// Same year
+		LocalDate testDateEndLast = LocalDate.parse("2017-08-12");
+		LocalDate testDateEndNext = LocalDate.parse("2017-11-23");
+		Map<String, String> periodMap = datesHelper.calculatePeriodRange(testDateEndLast, testDateEndNext, true);
+		assertEquals(null, periodMap.get(PERIOD_START));
```

Comment:
You can use assertNull(periodMap.get(PERIOD_START)) for null checks

Reply:
resolved in 9cfc864